janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -296,19 +296,30 @@ public void register(String store, String key, KeyInformation information, BaseT
                 if (map==Mapping.DEFAULT) map=Mapping.TEXT;
                 log.debug("Registering string type for {} with mapping {}", key, map);
                 mapping.field("type", "string");
+                String stringAnalyzer = (String) ParameterType.STRING_ANALYZER.findParameter(information.getParameters(), null);
+                String textAnalyzer = (String) ParameterType.TEXT_ANALYZER.findParameter(information.getParameters(), null);
                 switch (map) {
                     case STRING:
-                        mapping.field("index","not_analyzed");
+                        if(stringAnalyzer !=null){
```

Comment:
I think it is a good idea to derive/craft IDEA / Eclipse formatters and add them to source control here so everyone can use them. Created an issue https://github.com/JanusGraph/janusgraph/issues/242. Meanwhile, please just approximate the existing style in the file.